### PR TITLE
Append vendor names to keys of RTC responses from Vendors in Doubleclick

### PR DESCRIPTION
**Pull request by bradfrizzell**
Currently, there is nothing stopping a rogue vendor from purposefully returning the same key names as another vendor, which would then overwrite the other vendor's values when they get merged together.

This PR appends the vendor name to each key name in the RTC responses, to prevent this from happening.

### CODE REVIEW

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 1]
```diff
@@ -644,7 +645,9 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
           if (rtcResponse.response[key]) {
             this.jsonTargeting_[key] =
                 !!this.jsonTargeting_[key] ?
-                deepMerge(this.jsonTargeting_[key], rtcResponse.response[key]) :
+                deepMerge(this.jsonTargeting_[key],
+                    this.rewriteRtcKeys_(rtcResponse.response[key],
+                        rtcResponse.callout)) :
                 rtcResponse.response[key];
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: Don't you need to rewrite the keys here as well?  This should be:
if (rtcResponse.response[key]) {
   const rewrittenKeys = this.rewriteRtcKeys_(rtcResponse.response[key], rtcResponse.callout);
   this.jsonTargeting_[key] = !!this.jsonTargeting_[key] ? deepMerge(this.jsonTargeting_[key], rewrittenKeys) : rewrittenKeys);
}

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 2]
```diff
@@ -653,6 +656,26 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
...
+   * @return {!Object<string, string>}
+   * @private
+   */
+  rewriteRtcKeys_(response, callout) {
+    // Only perform this substitution for vendor-defined URLs.
+    if (!RTC_VENDORS[callout]) {
+      return response;
+    }
+    const newResponse = {};
+    let newKey;
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: Remove

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 3]
```diff
@@ -653,6 +656,26 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
...
+    Object.keys(response).forEach(key => {
+      newResponse[key + '_' + callout] = response[key];
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: Nit: Object.keys(response).forEach(key => newResponse[`${key}_${callout}`] = response[key]);

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js [hunk 4]
```diff
@@ -147,4 +147,28 @@ describes.realWin('DoubleClick Fast Fetch RTC', {amp: true}, env => {
...
+      const response = {
+        'a': '1',
+        'b': '2',
+      };
+      const rewrittenResponse = {
+        'a_fakevendor': '1',
+        'b_fakevendor': '2',
+      };
+      expect(impl.rewriteRtcKeys_(response, 'fakevendor')
+            ).to.deep.equal(rewrittenResponse);
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js]: Nit: move ) to previous line

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js [hunk 5]
```diff
@@ -147,4 +147,28 @@ describes.realWin('DoubleClick Fast Fetch RTC', {amp: true}, env => {
...
+    });
+
+    it('should not rewrite key names if custom url callout', () => {
+      const response = {
+        'a': '1',
+        'b': '2',
+      };
+      expect(impl.rewriteRtcKeys_(response, 'www.customurl.biz')
+            ).to.deep.equal(response);
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js]: ditto

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js (hunk 1 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: done

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js (hunk 2 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: done

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js (hunk 3 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: done

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js (hunk 4 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js]: done

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js (hunk 5 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js]: done

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js [hunk 6]
```diff
@@ -147,4 +147,28 @@ describes.realWin('DoubleClick Fast Fetch RTC', {amp: true}, env => {
     });
 
   });
+
+  describe('rewriteRtcKeys', () => {
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js]: I would have expected a need to change other tests in terms of their resulting json targeting, no?

**Pull request merged**